src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
+    string attributeFilePart = instanceDataFolderLocation + "/" + graphID + "_egonet_" + partitionID;
+    ofstream partfile;
+    partfile.open(attributeFilePart, std::fstream::trunc);
+    for (map<long, map<long, unordered_set<long>>>::iterator it = egonetMap.begin(); it != egonetMap.end(); ++it) {
+        map<long, unordered_set<long>> egonetInternalMap = it->second;
+        for (map<long, unordered_set<long>>::iterator itm = egonetInternalMap.begin(); itm != egonetInternalMap.end();
+             ++itm) {
+            unordered_set<long> egonetInternalMapEdges = itm->second;
+            for (unordered_set<long>::iterator ite = egonetInternalMapEdges.begin();
+                 ite != egonetInternalMapEdges.end(); ++ite) {
+                partfile << to_string(it->first) << "\t" << to_string(itm->first) << "\t" << to_string(*ite) << endl;
+            }
+        }
+    }
+    partfile.close();
 
-            write(connFd, JasmineGraphInstanceProtocol::SEND_PARTITION_COUNT.c_str(),
-                  JasmineGraphInstanceProtocol::SEND_PARTITION_ITERATION.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::SEND_PARTITION_COUNT, "info");
+    // todo  invoke other workers asynchronously
+    for (vector<string>::iterator workerIt = workerSockets.begin(); workerIt != workerSockets.end(); ++workerIt) {
+        instance_logger.log("Worker pair " + *workerIt, "info");
 
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string partCount(data);
+        std::vector<string> workerSocketPair;
+        stringstream wl(*workerIt);
+        string intermediate;
+        while (getline(wl, intermediate, ':')) {
+            workerSocketPair.push_back(intermediate);
+        }
 
-            instance_logger.log("Received partition iteration - " + partIteration, "info");
-            JasmineGraphInstanceService::collectExecutionData(partIteration, trainData, partCount);
-            instance_logger.log("After calling collector ", "info");
+        if (std::to_string(serverPort).compare(workerSocketPair[1]) == 0) {
+            continue;
+        }
 
-        } else if (line.compare(JasmineGraphInstanceProtocol::INITIATE_PREDICT) == 0) {
-            instance_logger.log("Received : " + JasmineGraphInstanceProtocol::INITIATE_PREDICT, "info");
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string graphID = (data);
-            graphID = Utils::trim_copy(graphID, " \f\n\r\t\v");
-            instance_logger.log("Received Graph ID: " + graphID, "info");
+        string host = workerSocketPair[0];
+        int port = stoi(workerSocketPair[1]);
+        int sockfd;
+        char data[301];
```

Comment:
```suggestion
        char data[INSTANCE_DATA_LENGTH + 1];
```

Reply:
Replaced all occurrences of `INSTANCE_DATA_LENGTH + 1` with `DATA_BUFFER_SIZE` where,
```
#define DATA_BUFFER_SIZE (INSTANCE_DATA_LENGTH + 1)
```

Reply:
Ok